src/types.rs

```diff
@@ -1751,6 +2562,24 @@ impl TRS {
     }
     /// Perform a single rewrite step using a normal-order (leftmost-outermost)
     /// rewrite strategy.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use term_rewriting::{Signature, TRS, parse_trs, Term, parse_term};
+    /// let mut sig = Signature::default();
+    ///
+    /// let t = parse_trs(&mut sig, 
+    /// "A = B;
+    /// C = D | E;
+    /// F(x_) = G;").expect("parse of A = B; C = D | E; F(x_) = G;");
+    ///
+    /// let mut term = parse_term(&mut sig, "J(A K(C A))").expect("parse of J(A K(C A))");
+    /// let expected_term = parse_term(&mut sig, "J(B K(C A))").expect("parse of J(B K(C A))");
+    ///
+    /// let rewriten_term = t.rewrite(&term).unwrap();
+    /// assert_eq!(rewriten_term, vec![expected_term]);
```

Comment:
This statement isn't very easy to scan if you're trying to get a sense for how the function behaves. Perhaps use something like `rewritten_term[0].pretty(&sig)` to convert the rewritten term to a string and then just write out the expected string?